### DISCUSSION TITLE
测试在Github Issues上写博客

### DISCUSSION THREAD

**Discussion by King-of-Infinite-Space** [King-of-Infinite-Space/thoughts#150] [存档]
找个地方写写东西记录想法的需求由来已久。主要需求如下：
1. 操作便捷
2. 用户有自主权
3. 运营稳定
4. 比较开放
5. 功能完善，支持分类、搜索等
5. 最好支持记录零碎想法（类似“微博”）
6. 最好支持多平台
7. 最好界面简洁美观

考虑过以下选择，然而各有各的问题。
- QQ空间（封闭落后、低龄）
- 微博（不符合个人喜好）
- 简书（正在使用，不过不符合6）
- 知乎（不符合6）
- B站（有点滑稽）
- 传统博客网站如新浪（落后）

此外国内平台普遍存在的问题是内容审查，所以一定程度上不满足2。
国外的知名平台比如blogger和medium都被墙，翻墙又带来了操作麻烦的问题。非知名平台又难以保证3。综上所述似乎剩下的出路不多，比如自己搭建。然而如何满足1？
此前了解到可以用Hexo和Github Pages来自建博客，不过每次发文都要在电脑上push的操作实在繁琐。
最近发现Github Issues可以当作博客来使用，基本满足以上所以条件。并且已有人开发在Github Pages上读取Issues生成博客的框架，这相当于为基于Github Issues的博客提供了一个更加美观的界面。所以我就选择了尝试这条路线。（当然，如果不太在意博客的界面，Github Issues本身就很好用了）
[这篇文章](https://www.jianshu.com/p/1cffcf9adbde)介绍了一些前面提到的框架，比较符合我需求的是[Spring](https://github.com/zhaoda/spring)，随后通过Github搜索我又发现了[BGAIssuesBlog](https://github.com/bingoogolapple/BGAIssueBlog)以及[vue-github-issues-blog](https://github.com/auven/vue-github-issues-blog)。不过它们没有提供完美的解决方案。第一个不支持读取Issues的评论，尽管有[修改版](https://github.com/f2e-journey/xueqianban)添加了显示评论功能，但是经我测试评论必须刷新页面后才能显示出来，应该是bug。第二个对移动端支持不好。第三个支持移动端，界面看起来也不错，但可惜似乎无法读取评论。以我自己完全业余的前端水平难以进行继续开发，于是只好选择比较好的将就着用。于是把前两个框架拿过来稍微做了些小修改并发布到了Github Pages上——[基于Spring修改版的版本](https://king-of-infinite-space.github.io/blog2/)、[基于BGAIssuesBlog的版本（主要）](https://king-of-infinite-space.github.io/blog/)，并新建了一个repo用来写issues。其中主要的那个版本里我将“微博”功能（本质上是在一篇文章里添加comments）做成了独立的标签，与其他文章分离。（顺便也接触了vue这一框架。虽然对其具体架构并不了解，但其大致思路还是可以领会一些，因而可以做些小的改动）

2018.8更新：
又发现了一个[稍复杂一些的](https://github.com/Monine/monine.github.io)。
<img src='https://count.lnfinite.space/post/1.svg?plus=1' width='0' height='0' />

**Comment by King-of-Infinite-Space** [comment]
Github真是伟大的网站。🐙🐱大法好。

**Comment by King-of-Infinite-Space** [comment]
2019.6更新
想把自己的博客修缮一下（或重建），正在评估各种方案。再次搜索并浏览了一下相关项目，发现了一些新项目或更新过的项目.
- [vue-github-issues-blog](https://github.com/auven/vue-github-issues-blog) 上面提到过的项目已更新，使用gitalk插件加载评论，效果不错
- [Mirror](https://github.com/LoeiFy/Mirror) 也不错，不过标签没有统一显示，且评论需要手动加载
- [gblog](https://github.com/ydq/gblog) 标签没有统一显示，不过有搜索功能
- [Issue-Blog](https://github.com/Yidadaa/Issue-Blog) 界面配图有艺术感。使用了milestone分类而label用来标记，也是一个启发。项目在积极更新中。

以上三者都可以满足我的需求：适应页面、分类、读取评论（可作为追加更新）。由于我的博客有一个分类当作微博来写，我希望 这些页面能倒序显示（最新评论在最上方）。搜索不是一个强需求，可以舍弃（毕竟可以在Issues的页面搜索）。从美观性的方面考虑倾向于最后一个（貌似是uestc的年轻人开发的，支持！）。现在要评估一下在各项目里实现“微博”功能的难度。

**Comment by King-of-Infinite-Space** [comment]
评论插件：gitalk, vssue
补：utteranc.es

**Comment by King-of-Infinite-Space** [comment]
续上：选用了[Issue-Blog](https://github.com/Yidadaa/Issue-Blog)，并做了些自定义。此前用的项目是单个页面，在运行时通过Github api抓取issue内容；而这一项目是先通过api获取内容并保存成md文件，然后通过VuePress渲染成静态html。这套流程通过Github Actions自动运行。第二步应该可以换成别的静态博客框架（比如hexo, Jekyll）。

最近有个想法——通过客户端把Github issues用作时间线样式的社交媒体。每个用户的内容存放在特定repo的issues里，其他用户star此repo即为关注。每条issue对应一条动态或一篇文章（可以通过milestone区分），label可以用来分类或当作hashtag。

注：如果在Github上follow某用户，那么对方的某些动态会在dashboard上显示。如果watch某repo，则该repo出现特定活动时会收到通知。

**Comment by King-of-Infinite-Space** [comment]
[Footnotes now supported in Markdown fields | GitHub Changelog](https://github.blog/changelog/2021-09-30-footnotes-now-supported-in-markdown-fields/)

test 1 [^1]
test 1 [^2]
test 3 [^3]
test 1 again [^1]

[^1]: ref1
[^2]: ref2

**Comment by King-of-Infinite-Space** [comment]
[giscus](https://giscus.app/) 类似之前提到的评论插件，但这个是基于GitHub Discussions而非Issues。

**Comment by King-of-Infinite-Space** [comment]
Github Markdown now supports LaTex through MathJax
[Render mathematical expressions in Markdown | GitHub Changelog](https://github.blog/changelog/2022-05-19-render-mathematical-expressions-in-markdown/)

test $inline$ math
$$standalone$$

**Comment by shenweiyan** [comment]
GitHub Discussions 大法好，再配合 [giscus](https://giscus.app/)，从博文到评论都能一一展示，剩下的专注码字就可以。
